resources/aws/rti_aws_uploader.py

```diff
@@ -0,0 +1,264 @@
+                for obj_summary in bucket.objects.all()]
+    if obj_list:
+        print('Listing bucket Objects:')
+        for obj in obj_list:
+            print(f'Obj ({obj.last_modified}) {obj.key}')
+    else:
+        print('The bucket is empty.')
+
+
+def _get_directory_tree_file_path_list(
+        top_dir: Path, filtered_out_subdirs: Set[str] = None) -> List[Path]:
+    """Obtains directory tree structure.
+
+    Args:
+        top_dir (Path): top directory path.
+        filtered_out_subdirs (Set[str]): filtered out subdir list
+            (default=None).
+
+    Returns:
+        file_list (List[Path]): all file paths of the tree.
+    """
+    file_list: List[Path] = []
+    for curr_path, _, files in os.walk(top_dir):
+        dir_path = Path(curr_path).resolve()
+        if (filtered_out_subdirs and
+                filtered_out_subdirs.intersection(dir_path.parts)):
+            continue
+        for curr_file in files:
+            file_list.append(dir_path.joinpath(curr_file).resolve())
+    return file_list
+
+
+def _get_filtered_file_list(
+        top_dir: Path,
+        selected_top_items: Set[str] = None,
+        filtered_out_subdirs: Set[str] = None) -> List[Path]:
+    """Obtains a list of files under top_dir with their own paths.
+
+    Args:
+        top_dir (Path): top directory path.
+        selected_top_items (Set[Path]): selected top items list (default=None).
+        filtered_out_subdirs (Set[Path]): filtered out subdir list
+            (default=None).
+
+    Returns:
+        Sorted file path list.
+    """
+    file_list: List[Path] = []
+    for child in top_dir.iterdir():
+        if selected_top_items and child.name not in selected_top_items:
+            continue
+        if child.is_dir():
+            file_list.extend(
+                _get_directory_tree_file_path_list(child.resolve(),
+                                                   filtered_out_subdirs))
+        else:
+            file_list.append(child.resolve())
+    return sorted(file_list)
+
+
+def upload_directory(s3,
+                     bucket_name: str,
+                     local_dir_path: Path,
+                     selected_top_items: Set[str],
+                     filtered_out_subdirs: Set[str]) -> None:
+    """Uploads local directory tree to the bucket.
+
+    Args:
+        s3 (boto3.resource('s3')): s3 resource.
+        bucket_name (str): bucket name.
+        local_dir_path (Path): directory path to upload.
+        selected_top_items (Set[Path]): selected top items list (default=None).
+        filtered_out_subdirs (Set[Path]): filtered out subdir list
+            (default=None).
+    """
+    bucket = s3.Bucket(bucket_name)
+    file_path_filtered_list = _get_filtered_file_list(
+        local_dir_path, selected_top_items, filtered_out_subdirs)
+    print('The following files will be uploaded:')
+    print(*file_path_filtered_list, sep='\n')
+    if input('Are you sure? (yes/no): ').strip() != 'yes':
```

Comment:
This should not go here. Why? If I am using the `upload_directory` function, it will be totally unexpected to be blocked by an user interaction.